event/src/unvalidated/event.rs

```diff
@@ -119,6 +119,19 @@ where
                     .ok_or_else(|| anyhow!("Signed Event CAR data missing block for payload"))?;
                 let payload =
                     serde_ipld_dagcbor::from_slice(payload_bytes).context("decoding payload")?;
+                let capability = envelope
+                    .capability()
+                    .map(|capability_cid| -> anyhow::Result<_> {
+                        let capability_bytes =
+                            car_blocks.get(&capability_cid).ok_or_else(|| {
+                                anyhow!("Signed Event CAR data missing block for capability")
+                            })?;
+                        let capability: signed::cacao::Capability =
+                            serde_ipld_dagcbor::from_slice(capability_bytes)
+                                .context("decoding capability")?;
+                        Ok((capability_cid, capability))
+                    })
+                    .transpose()?;
```

Comment:
what is `transpose()` doing?  I'm having a hard time googling for any docs on it.

Reply:
Its this https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose